kythe/web/site/install-macos.md

```diff
@@ -0,0 +1,110 @@
+---
+layout: page
+title: Getting Started on macOS
+permalink: /getting-started-macos/
+---
+
+* toc
+{:toc}
+
+This document extends the [Getting Started guide](getting-started.md) to cover
+installation of external dependencies on a macOS system. You will still need to
+follow all the other Getting Started instructions to get a working Kythe build.
+
+
+## Installing Developer Tools
+
+**Xcode**: To build on macOS, you will need Xcode installed. You can do this
+from the terminal by running:
+
+{% highligh bash %}
+softwareupdate --install Xcode
+{% endhighlight %}
+
+or by launching the App Store application and browsing to Develop → Xcode.
+
+**Homebrew**: You will also need to install [Homebrew](https://brew.sh) (follow
+the link for instructions).  There are other ways to install packages, but the
+rest of these instructions assume you have it.
+
+
+## Installing External Dependencies
+
+To install most of the [external dependencies][ext], run
+
+{% highlight bash %}
+for pkg in asciidoc cmake go graphviz node parallel source-highlight ; do
+   brew install $pkg
+done
+
+# Bazel.
+# See also:  https://docs.bazel.build/versions/master/install-os-x.html
+#
+# DO NOT run "brew install bazel"; the version from core does not work.
+# If you did so by mistake, run "brew uninstall bazel" first.
+brew tap bazelbuild/tap
+brew tap-pin bazelbuild/tap
+brew install bazelbuild/tap/bazel
+
+# Docker: See the instructions below.
+# DO NOT use brew install docker (or if you did: brew uninstall docker).
+
+# Optional installs:
+brew install leiningen
+brew install ninja
+{% endhighlight %}
+
+Note that some of the external packages listed are not mentioned here, because
+macOS already has suitable versions pre-installed.
+
+**Docker**: Download and install Docker for Mac from the [docker store][dock].
+Note that you will have to create a "Docker ID" and log in to get access to the
+download, but you won't to log in to the store to use Docker once it has been
+installed.
+
+
+## Verifying the Installation
+
+Once you have finished setting up your macOS installation, you can verify that
+everything works by running:
+
+{% highlight bash %}
+cd $KYTHE_DIR
```

Comment:
no need to run tools/modules/update.sh ?

Reply:
See above—this is _after_ you have installed everything else.

Reply:
when is the user expected to run update.sh? Is that from the previous page that linked to this page?

Reply:
Ah if tools/modules/update is expected to be run before someone even enters this doc, then sure.

Reply:
But I'd think that you can't run ~anything before you install xcode and bazel?

Reply:
Ah good point, tools/modules/update runs bazel build //tools/modules:compiler_info

Reply:
I attempted to clarify the language a bit. Or if you prefer I could (a) drop it entirely, or (b) move it to the main document.

Reply:
Sure that looks reasonable.  Maybe include a pointer back to the main docs from "verifying the installation", just for ease of flow?

Reply:
Sure, sounds good—done!